File: apps/api/app/Models/MonitorPackage.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class MonitorPackage extends Model
{
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'commerce_id',
        'package_name',
        'app_name',
        'description',
        'is_active',
        'enabled_default',
        'priority',
    ];

    /**
     * Get the attributes that should be cast.
     *
     * @return array<string, string>
     */
    protected function casts(): array
    {
        return [
            'is_active' => 'boolean',
            'enabled_default' => 'boolean',
            'priority' => 'integer',
        ];
    }

    /**
     * Scope a query to only include active packages.
     */
    public function scopeActive($query)
    {
        return $query->where('is_active', true);
    }

    /**
     * Scope a query to order by priority.
     */
    public function scopeOrdered($query)
    {
        return $query->orderBy('priority', 'desc')->orderBy('package_name', 'asc');
    }

    /**
     * Get the commerce for this monitor package.
     */
    public function commerce(): BelongsTo
    {
        return $this->belongsTo(Commerce::class);
    }
}

File: apps/android-client/app/src/main/java/com/yapenotifier/android/data/model/MonitorPackage.kt
package com.yapenotifier.android.data.model

import com.google.gson.annotations.SerializedName

/**
 * Represents a monitor package (app to monitor for payment notifications).
 */
data class MonitorPackage(
    @SerializedName("id")
    val id: Long,

    @SerializedName("commerce_id")
    val commerceId: Long?,

    @SerializedName("package_name")
    val packageName: String,

    @SerializedName("app_name")
    val appName: String?,

    @SerializedName("description")
    val description: String?,

    @SerializedName("is_active")
    val isActive: Boolean,

    @SerializedName("enabled_default")
    val enabledDefault: Boolean?,

    @SerializedName("priority")
    val priority: Int?,

    @SerializedName("created_at")
    val createdAt: String?,

    @SerializedName("updated_at")
    val updatedAt: String?
)

